juju_verify/utils/unit.py

```diff
@@ -76,13 +106,13 @@ def run_action_on_units(
              provided in 'units' and actions are their matching,
              juju.Action objects that have been executed and awaited.
     """
-    with cache_manager(use_cache):
-        task_map = {unit.entity_id: run_action(unit, action, params) for unit in units}
+    task_map = {
+        unit.entity_id: run_action(unit, action, params, use_cache) for unit in units
```

Comment:
This returns:
```
/snap/juju-verify/x11/lib/python3.8/site-packages/juju_verify/utils/unit.py:110: RuntimeWarning: coroutine 'run_action' was never awaited
  unit.entity_id: run_action(unit, action, params, use_cache) for unit in units
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

Reply:
This is not related to these changes and I proposed fix for it in #59. Thanks